x-pack/test/rule_registry/spaces_only/tests/trial/lifecycle_executor.ts

```diff
@@ -0,0 +1,242 @@
+      const ruleDataService = new RuleDataService({
+        getClusterClient,
+        logger,
+        kibanaVersion: '8.0.0',
+        isWriteEnabled: true,
+        isWriterCacheEnabled: false,
+        disabledRegistrationContexts: [] as string[],
+      });
+
+      // This initializes the service. This happens immediately after the creation
+      // of the RuleDataService in the setup phase of the Rule Registry plugin
+      ruleDataService.initializeService();
+
+      // This initializes the index and templates and returns the data client.
+      // This happens in each solution plugin before they can register lifecycle
+      // executors.
+      ruleDataClient = ruleDataService.initializeIndex({
+        feature: AlertConsumers.OBSERVABILITY,
+        registrationContext: 'observability.test',
+        dataset: Dataset.alerts,
+        componentTemplateRefs: [],
+        componentTemplates: [
+          {
+            name: 'mappings',
+            mappings: mappingFromFieldMap(
+              {
+                testObject: {
+                  type: 'object',
+                  required: false,
+                  array: false,
+                },
+              },
+              false
+            ),
+          },
+        ],
+      });
+    });
+
+    after(async () => {
+      cleanupRegistryIndices(getService, ruleDataClient);
+    });
+
+    it('should work with object fields', async () => {
+      const id = 'host-01';
+
+      // This creates the function that will wrap the solution's rule executor with the RuleRegistry lifecycle
+      const createLifecycleRuleExecutor = createLifecycleExecutor(logger, ruleDataClient);
+
+      // This creates the executor that is passed to the Alerting framework.
+      const executor = createLifecycleRuleExecutor<
+        MockRuleParams,
+        MockRuleState,
+        MockAlertState,
+        MockAlertContext,
+        MockAllowedActionGroups
+      >(async function (options) {
+        const { services, state: previousState } = options;
+        const { alertWithLifecycle } = services;
+
+        // Fake some state updates
+        const state = previousState.testObject
+          ? {
+              ...previousState,
+              testObject: {
+                ...previousState.testObject,
+                values: [
+                  ...previousState.testObject.values,
+                  { name: 'count', value: previousState.testObject.values.length + 1 },
+                ],
+              },
+            }
+          : {
+              ...previousState,
+              testObject: {
+                id,
+                values: [{ name: 'count', value: 1 }],
+                host: {
+                  name: id,
+                },
+              },
+            };
+
+        // This MUST be called by the solutions executor function
+        alertWithLifecycle({
+          id,
+          fields: {
+            [ALERT_REASON]: 'Test alert is firing',
+            ...state,
```

Comment:
@simianhacker Can you explain maybe a bit more what you are doing here? I had the impression that we are calling alertWithLifecycle we pass we want to index https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts#L102. So you index a new key called `testObject`? Could you clarify?

Reply:
Yes, spreading the state will add the `testObject` to the index. When I was poking around in different solutions, I noticed that each plugin used `alertWithLifecycle` to add their additional metadata.